### PR TITLE
update maintainers periodic prow job

### PR DESCRIPTION
**Pull request by leodido**
This PR introduces:

- A ProwJob ("update-maintainers") that will periodically:
    - check all the OWNERS files present in the **falcosecurity** organization (using [leodido/maintainers-generator](https://github.com/leodido/maintainers-generator))
    - create  the current list of maintainers of all the Falco projects in a YAML file (using [leodido/maintainers-generator](https://github.com/leodido/maintainers-generator))
    - open a PR towards **falcosecurity/.github** repository on the ([maintainers.yaml](https://github.com/falcosecurity/.github/blob/master/maintainers.yaml)) file, in case there are updates
- Improvements to the scripts inside `tools/` to run ProwJob locally
- Improvements to the guide to run ProwJob locally

TODOs:

- [x] Final testing
- [x] Creation of the test-infra/update-maintainers repository on AWS ECR registry
- [x] Job to build `update-maintainers` container image

### CODE REVIEW

File: images/update-maintainers/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+set -o nounset
+set -o pipefail
+
+# Args from environment (with defaults)
+GH_ORG="${GH_ORG:-"falcosecurity"}"
+GH_REPO="${GH_REPO:-".github"}"
+BOT_NAME="${BOT_NAME:-"poiana"}"
+BOT_MAIL="${BOT_MAIL:-"leo+bot@sysdig.com"}"
+BOT_GPG_KEY_PATH="${BOT_GPG_KEY_PATH:-"/root/gpg-signing-key/poiana.asc"}"
+BOT_GPG_PUBLIC_KEY="${BOT_GPG_PUBLIC_KEY:-"5B969CD19422B477E5609F8C900C09B3E21C193F"}"
```
Comment (jonahjon) [inline images/update-maintainers/entrypoint.sh:24]: Is it ok to put the pub key in here?

**Review by jonahjon — APPROVED** [review]

**Comment by jonahjon** [comment]
Maybe also add a job for building the `update-maintainers` docker image

File: images/update-maintainers/entrypoint.sh (hunk 1 above)
Comment (leodido) [reply images/update-maintainers/entrypoint.sh:24]: It's like having it [here](https://github.com/falcosecurity/test-infra/blob/master/config/jobs/autobump/autobump.yaml#L23)

I just put it as a default :)

Comment (jonahjon) [reply images/update-maintainers/entrypoint.sh:24]: O true

File: config/jobs/update-maintainers/update-maintainers.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      workdir: true
+    spec:
+      containers:
+      # See images/update-maintainers
+      - image: 292999226676.dkr.ecr.eu-west-1.amazonaws.com/test-infra/update-maintainers # Local testing: substite with => image: localhost:5000/update-maintainers
+        imagePullPolicy: Always
+        command:
+        - entrypoint.sh # Local testing: sleep
+        args:
+        -  /etc/github-token/oauth # Local testing: substite with => - "900"
```
Comment (leodido) [inline config/jobs/update-maintainers/update-maintainers.yaml:20]:
```suggestion
        -  /etc/github-token/oauth # Local testing: substitute with => - "900"
```

File: config/jobs/update-maintainers/update-maintainers.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    # Check out the repo containing the maintainers file
+    # This will be the working directory
+    - org: falcosecurity
+      repo: .github
+      base_ref: master
+      workdir: true
+    spec:
+      containers:
+      # See images/update-maintainers
+      - image: 292999226676.dkr.ecr.eu-west-1.amazonaws.com/test-infra/update-maintainers # Local testing: substite with => image: localhost:5000/update-maintainers
```
Comment (leodido) [inline config/jobs/update-maintainers/update-maintainers.yaml:15]:
```suggestion
      - image: 292999226676.dkr.ecr.eu-west-1.amazonaws.com/test-infra/update-maintainers # Local testing: substitute with => image: localhost:5000/update-maintainers
```

**Review by jonahjon — APPROVED** [review]

**Pull request merged**